chapter_attention-mechanisms/attention-scoring-functions.ipynb

```diff
@@ -0,0 +1,634 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "System.out.println(\n",
+    "        maskedSoftmax(\n",
+    "                manager.randomUniform(0, 1, new Shape(2, 2, 4)),\n",
+    "                manager.create(new float[] {2, 3})));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "origin_pos": 9
+   },
+   "source": [
+    "Similarly, we can also\n",
+    "use a two-dimensional NDArray\n",
+    "to specify valid lengths\n",
+    "for every row in each matrix example.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "System.out.println(\n",
+    "        maskedSoftmax(\n",
+    "                manager.randomUniform(0, 1, new Shape(2, 2, 4)),\n",
+    "                manager.create(new float[][] {{1, 3}, {2, 4}})));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "origin_pos": 12
+   },
+   "source": [
+    "## Additive Attention\n",
+    ":label:`subsec_additive-attention`\n",
+    "\n",
+    "In general,\n",
+    "when queries and keys are vectors of different lengths,\n",
+    "we can use additive attention\n",
+    "as the scoring function.\n",
+    "Given a query $\\mathbf{q} \\in \\mathbb{R}^q$\n",
+    "and a key $\\mathbf{k} \\in \\mathbb{R}^k$,\n",
+    "the *additive attention* scoring function\n",
+    "\n",
+    "$$a(\\mathbf q, \\mathbf k) = \\mathbf w_v^\\top \\text{tanh}(\\mathbf W_q\\mathbf q + \\mathbf W_k \\mathbf k) \\in \\mathbb{R},$$\n",
+    ":eqlabel:`eq_additive-attn`\n",
+    "\n",
+    "where\n",
+    "learnable parameters\n",
+    "$\\mathbf W_q\\in\\mathbb R^{h\\times q}$, $\\mathbf W_k\\in\\mathbb R^{h\\times k}$, and $\\mathbf w_v\\in\\mathbb R^{h}$.\n",
+    "Equivalent to :eqref:`eq_additive-attn`,\n",
+    "the query and the key are concatenated\n",
+    "and fed into an MLP with a single hidden layer\n",
+    "whose number of hidden units is $h$, a hyperparameter.\n",
+    "By using $\\tanh$ as the activation function and disabling\n",
+    "bias terms,\n",
+    "we implement additive attention in the following.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "/* Additive attention. */\n",
+    "public class AdditiveAttention extends AbstractBlock {\n",
+    "    private static final byte VERSION = 1;\n",
+    "    private Linear W_k;\n",
+    "    private Linear W_q;\n",
+    "    private Linear W_v;\n",
+    "    private Dropout dropout;\n",
+    "    public NDArray attentionWeights;\n",
+    "\n",
+    "    public AdditiveAttention(int numHiddens, float dropout) {\n",
+    "        super(VERSION);\n",
+    "        this.W_k = Linear.builder().setUnits(numHiddens).optBias(false).build();\n",
+    "        this.W_k.setInitializer(new UniformInitializer(0.07f), Parameter.Type.WEIGHT);\n",
```

Comment:
I checked the original book. they seems to use default initializer.

Reply:
I think I had specified because I had seen it used Uniform when debugging compared to Xavier which is default for weights on DJL. I removed and it still brings same result so I'll change it.